express-delete/index.js

```diff
@@ -0,0 +1,40 @@
+const express = require('express');
+const app = express();
+app.listen(3000, () => 'server is listening');
+
+const grades = {
+  12: {
+    id: 12,
+    name: 'Tim Berners-Lee',
+    course: 'HTML',
+    score: 95
+  },
+  47: {
+    id: 47,
+    name: 'Brendan Eich',
+    course: 'JavaScript',
+    score: 100
+  },
+  273: {
+    id: 273,
+    name: 'Forbes Lindsay',
+    course: 'JavaScript',
+    score: 92
+  }
+};
+
+app.get('/api/grades', (req, res, next) => {
+  const arr = [];
+  for (const gr in grades) {
+    arr.push(grades[gr]);
+  }
+  res.json(arr);
+  next();
+});
+
+app.delete('/api/grades/:id', (req, res, next) => {
+  const id = req.params.id;
+  delete grades[id];
+  res.sendStatus(204);
+  next();
+});
```

Comment:
If you've sent a response to the client and don't have additional work to do on the server, avoid calling `next()`. It can cause additional, unexpected logic to run, including other routes if the path just happens to match.